### ISSUE TITLE
Using apply_transform with PyTorch

### ISSUE THREAD

**Issue by CarlArblue** [lululxvi/deepxde#601]
Hello

So I have been trying to use apply_transform with PyTorch. I can get the model to compile but the loss does not decrease. I believe there is an error in how I have create the sequential model in PyTorch. Perhaps you are able to tell me where my error is? I have supplied the code below:

# ZETA LAYERS
zeta_layer_size = 15
pytorch_zeta_call = torch.nn.Sequential(
    torch.nn.Flatten(),
    torch.nn.LazyLinear(out_features=zeta_layer_size, bias=False),
    torch.nn.Tanh(),
    torch.nn.LazyLinear(out_features=zeta_layer_size, bias=False),
    torch.nn.Tanh(),
    torch.nn.LazyLinear(out_features=zeta_layer_size, bias=False),
    torch.nn.Tanh(),
    torch.nn.LazyLinear(out_features=1, bias=False)
    )

# PHI LAYERS
phi_layer_size = 20
pytorch_phi_call = torch.nn.Sequential(
    torch.nn.Flatten(),
    torch.nn.LazyLinear(out_features=phi_layer_size, bias=False),
    torch.nn.Tanh(),
    torch.nn.LazyLinear(out_features=phi_layer_size, bias=False),
    torch.nn.Tanh(),
    torch.nn.LazyLinear(out_features=phi_layer_size, bias=False),
    torch.nn.Tanh(),
    torch.nn.LazyLinear(out_features=1, bias=False)
    )

def output_transform(input, outputs):
    x = input[:,0:1]
    z = input[:,1:2]
    t = input[:,2:3]

    x_t = torch.concat([x,t], axis=1)
    x_z_t = torch.concat([x, z, t], axis=1)

    zeta_output = pytorch_zeta_call(x_t)
    phi_output = pytorch_phi_call(x_z_t)

    # FINAL OUTPUT #
    final_output = torch.concat([zeta_output, phi_output], axis=1)

    return final_output

def input_transform(input):
    x = input[:,0:1]
    sigma = input[:,1:2]
    t = input[:,2:3]

    x_t = torch.concat([x,t], axis=1)

    zeta = pytorch_zeta_call(x_t)

    z = sigma*(zeta+hd)-hd
    # z = tf.stop_gradient(z)
    
    # FINAL INPUT #
    final_input = torch.concat([x, z, t], axis=1)
    
    return final_input

FNN_layer = [3,2] # x, sigma, t -> zeta, phi
net = dde.nn.FNN(FNN_layer, "tanh", "Glorot uniform")
net.apply_feature_transform(input_transform)
net.apply_output_transform(output_transform)
model = dde.Model(data, net)

Thanks in advance.

**Comment by lululxvi** [comment]
See example: https://github.com/lu-group/sbinn
